pytorch_lightning/core/lightning.py

```diff
@@ -1969,3 +1969,183 @@ def _register_sharded_tensor_state_dict_hooks_if_available(self) -> None:
 
         self._register_state_dict_hook(state_dict_hook)
         self._register_load_state_dict_pre_hook(pre_load_state_dict_hook, True)
+
+    def init_optimizers_and_lr_schedulers(self) -> Tuple[List, List, List]:
```

Comment:
should this be private?

also quick ques: why lightningModule is handling the parsing/validation logic?

Reply:
> should this be private?

Since it is used by other classes across our codebase I initially left it as public but since we don't want users to access it I have now made it private. This is a problem we have with many private methods in the codebase, where they are not truly private...

> why lightningModule is handling the parsing/validation logic?

I agree with @awaelchli 's reasoning in https://github.com/PyTorchLightning/pytorch-lightning/issues/10681#issuecomment-997146642: "where it's close to the hook that retrieves the optimizers and schedulers from the user (configure_optimizers)"

But I am open to other suggestions of where to put this logic!

Reply:
A collection of possible places I can come up with:
- core/lightning.py
- core/optimizer.py
- pl/utilities/parsing??.py

Reply:
+1 for core/optimizers, maybe as a function?

Reply:
Moved it to `core/optimizers` as functions!